spec/jobs/reports/irs_verification_report_spec.rb

```diff
@@ -1,184 +1,146 @@
+# spec/jobs/reports/sp_verification_report_spec.rb
 require 'rails_helper'
-require 'active_support/testing/time_helpers'
+require 'csv'
 
-RSpec.describe Reports::IrsVerificationReport do
-  include ActiveSupport::Testing::TimeHelpers
+RSpec.describe Reports::SpVerificationReport do
+  let(:report_date) { Time.zone.parse('2025-11-14 23:59:59 UTC') } # Friday
+  let(:internal_emails) { ['internal1@example.com', 'internal2@example.com'] }
+  let(:partner_emails)  { ['partner@example.com'] }
+  let(:issuers)         { ['issuer1', 'issuer2'] }
+  let(:agency)          { 'Test_partner' }
 
-  let(:report_date) { Time.zone.today.end_of_day }
-  let(:receiver) { :internal }
-  let(:report) { described_class.new(report_date, receiver) }
-  let(:dummy_report_data) { [['Header1', 'Header2'], ['Value1', 'Value2']] }
-  let(:mock_test_irs_verification_emails) do
-    ['mock_feds@example.com', 'mock_contractors@example.com']
-  end
-  let(:mock_test_internal_emails) { ['mock_internal@example.com'] }
-
-  let(:mock_report_object) do
-    instance_double(
-      Reporting::IrsVerificationReport,
-      as_emailable_reports: [
-        Struct.new(:table, :filename).new(dummy_report_data, 'dummy.csv'),
-      ],
-    )
+  let(:configs) do
+    [
+      {
+        'issuers' => issuers,
```

Comment:
let's rename the values as "test_issuers" or "mock_issuers" etc. 

It might also be worth having multiple versions of these so you can test strings vs string arrays